packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -429,16 +498,24 @@ export class RunningSummarizer implements IDisposable {
 		cancellationToken = this.cancellationToken,
 		resultsBuilder = new SummarizeResultBuilder(),
 	): ISummarizeResults {
-		this.lockedSummaryAction(async () => {
-			const summarizeResult = this.generator.summarize(
-				summarizeProps,
-				options,
-				cancellationToken,
-				resultsBuilder,
-			);
-			// ensure we wait till the end of the process
-			return summarizeResult.receivedSummaryAckOrNack;
-		}).catch((error) => {
+		this.lockedSummaryAction(
+			async () => {
+				const summarizeResult = this.generator.summarize(
+					summarizeProps,
+					options,
+					cancellationToken,
+					resultsBuilder,
+				);
+				// ensure we wait till the end of the process
+				return summarizeResult.receivedSummaryAckOrNack;
+			},
+			() => {
```

Comment:
Just curious, why not do the before and after directly at the call site instead of inside the lockedSummaryAction function?

Reply:
no particular reason. the idea was to keep existing behavior of the lockdSummaryAction.